tests/error-display/error-display.spec.tsx

```diff
@@ -0,0 +1,322 @@
+
+            renderComponent({
+                type,
+                additionalProps,
+            });
+
+            const error = ERROR_DISPLAY_DATA.get(type);
+            const errorDescription = transformJSXElementToString(
+                error.renderDescription(additionalProps) as JSX.Element
+            );
+
+            expect(
+                screen.getByTestId(ERROR_DESCRIPTION_TEST_ID).textContent
+            ).toBe(errorDescription);
+        });
+    });
+
+    describe("unsupported browser error", () => {
+        it("should be able to render", () => {
+            const type = "unsupported-browser";
+            renderComponent({ type });
+
+            const error = ERROR_DISPLAY_DATA.get(type);
+            expect(
+                screen.getByRole("heading", { level: 1, name: error.title })
+            ).toBeInTheDocument();
+
+            expect(screen.getByRole("img", { name: "error" })).toHaveAttribute(
+                "src",
+                error.img.src
+            );
+            expect(
+                screen.getByTestId(ERROR_DESCRIPTION_TEST_ID).textContent
+            ).toBe(error.description);
+        });
+    });
+
+    describe("partially supported browser error", () => {
+        it("should be able to render", () => {
+            const type = "partially-supported-browser";
+            renderComponent({ type });
+
+            const error = ERROR_DISPLAY_DATA.get(type);
+            expect(
+                screen.getByRole("heading", { level: 1, name: error.title })
+            ).toBeInTheDocument();
+
+            expect(screen.getByRole("img", { name: "error" })).toHaveAttribute(
+                "src",
+                error.img.src
+            );
+            expect(
+                screen.getByTestId(ERROR_DESCRIPTION_TEST_ID).textContent
+            ).toBe(error.description);
+        });
+    });
+});
+
+// =============================================================================
+// CONSTANTS
+// =============================================================================
+const CUSTOM_TITLE = "custom error";
+const CUSTOM_DESCRIPTION = "custom description";
+const ERROR_DESCRIPTION_TEST_ID = "error-display--description";
+
+// =============================================================================
+// HELPER FUNCTIONS
+// =============================================================================
+const transformJSXElementToString = (element: JSX.Element): string => {
+    let text = "";
+
+    element.props.children.forEach((child) => {
+        if (typeof child === "object") {
+            text += child.props.children;
+        } else {
+            text += child;
+        }
+    });
+
+    return text;
+};
+
+// =============================================================================
+// RENDER FUNCTIONS
+// =============================================================================
+const renderComponent = (props?: Partial<ErrorDisplayProps>) => {
```

Comment:
Able to just render the component in the test. This makes it easier to read. Not alot of code anyway. Same for the others